src/lib.rs

```diff
@@ -113,7 +117,7 @@ async fn handle(
     }
 }
 
-async fn spawn_fetch_campaigns(market_uri: &str) -> Result<Cache, reqwest::Error> {
+async fn spawn_fetch_campaigns(market_uri: &String) -> Result<Cache, reqwest::Error> {
```

Comment:
Because of the way Rust works, you don't need `&String`, you can leave it as `&str` and pass a `....(&someStringVar)` which will automatically use the `&String` as a `&str`